### PR TITLE
chore: integrate visual-regression in CI

### PR DESCRIPTION
**Pull request by andreapiras**
## 🖼 Context

This Pr improves the developer experience by easing the visual-regression process. We don't have at the moment an official tool for testing so the integration cannot be fully automate atm

## 🚀 Changes

Add chromatic optional test that will generate a test build. We don't have an integration yet with github because of security reasons. Please do use consciously as for now we do have a limited plan 
image

## ✅ Checklist

- [ ] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [ ] Changes are tested and all tests pass
- [ ] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [ ] Changes work in all supported browsers (don't forget IE11)
- [ ] You've updated the documentation if necessary

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -156,6 +167,26 @@ workflows:
...
+          requires:
+            - test
+            - setup
+          filters:
+            branches:
+              ignore:
+                - gh-pages
+                - main
+
+      - visual-regression:
```
Comment (dozortsev) [inline .circleci/config.yml:181]: why we need two jobs here? can't it be just `visual-regression`

File: packages/components/CONTRIBUTING.md [hunk 2]
```diff
@@ -56,6 +56,12 @@ Before submitting a PR you should ensure:
 
 If you've added a new component, it's a good idea to try using it in a fresh create react app / nextjs environment. To do this you'll need to make use of `yarn link`, to ensure the app depends on your local code and not the last published package.
 
+### Visual Regression testing.
+We haven't integrated yet a visual testing tool officially but it's possible to run a visual regression manually for your branch changes to increase your confidence in the release.
+The regression is integrated with CI in order to run your test:
+- open a PR
+- manually approve the visual regression check at the end of the PR page.
```
Comment (dozortsev) [inline packages/components/CONTRIBUTING.md:63]: I would make list here, just:

```
The regression is integrated with CI flow in order to run your test, you just have to manually approve `visual-regression` job once to run the job.
```

File: packages/components/package.json [hunk 3]
```diff
@@ -37,7 +37,9 @@
     "lint": "eslint 'src/**/*.js'",
     "lint:fix": "yarn lint --fix",
     "lintless": "stylelint ./src/**/*.less",
-    "lintless:fix": "yarn lintless --fix"
+    "lintless:fix": "yarn lintless --fix",
+    "build-storybook": "yarn build && build-storybook -c .storybook -o ./dist/my-component/storybook/",
+    "visual-regression": "yarn chromatic CHROMATIC_PROJECT_TOKEN || true"
```
Comment (dozortsev) [inline packages/components/package.json:42]: maybe let's call `visual-test`? it's about tests, right? it's more concise and consistent with already existing `test` jobs and commands

File: packages/components/CONTRIBUTING.md [hunk 4]
```diff
@@ -56,6 +56,12 @@ Before submitting a PR you should ensure:
 
 If you've added a new component, it's a good idea to try using it in a fresh create react app / nextjs environment. To do this you'll need to make use of `yarn link`, to ensure the app depends on your local code and not the last published package.
 
+### Visual Regression testing.
+We haven't integrated yet a visual testing tool officially but it's possible to run a visual regression manually for your branch changes to increase your confidence in the release.
```
Comment (dozortsev) [inline packages/components/CONTRIBUTING.md:60]: I was a bit confused at when first read this sentence, suggesting clarify things a bit:

```
... but for now it's possible to run a visual test using Cromatic for your changes as part of your PR to increase your confidence in the release.
```

File: packages/components/CONTRIBUTING.md [hunk 5]
```diff
@@ -56,6 +56,12 @@ Before submitting a PR you should ensure:
...
+The regression is integrated with CI in order to run your test:
+- open a PR
+- manually approve the visual regression check at the end of the PR page.
+
```
Comment (dozortsev) [inline packages/components/CONTRIBUTING.md:64]: what would next steps for me as a contributor? where can I link on Chromatic results etc? could you please info about that here?

**Review by dozortsev — COMMENTED** [review]
Really nice!

File: packages/components/package.json (hunk 3 above)
Comment (andreapiras) [reply packages/components/package.json:42]: How about test:visual?

File: .circleci/config.yml (hunk 1 above)
Comment (andreapiras) [reply .circleci/config.yml:181]: I know this is weird but it works like this, you need a job to `hold` and then once you approve the other one that was hold is going to be triggered

Comment (andreapiras) [reply .circleci/config.yml]: https://circleci.com/blog/manual-job-approval-and-scheduled-workflow-runs/

Comment (dozortsev) [reply .circleci/config.yml]: no 🙂

it should just have `type: approval` prop https://circleci.com/docs/2.0/workflows/#holding-a-workflow-for-a-manual-approval

I tested this while i18n RFC https://app.circleci.com/pipelines/github/transferwise/neptune-web/5336/workflows/72e0c371-d39f-44cf-8933-20859a0c6b00

**Review by dozortsev — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (andreapiras) [reply .circleci/config.yml]: well we discovered you were totally wrong 😂

**Pull request merged**